File: lib/models/membership_plan.dart
class MembershipPlan {
  final String id;
  final String name;
  final String tier;
  final int monthlyPriceCents;
  final int yearlyPriceCents;
  final String currency;
  final bool isFamilyPlan;
  final int? maxFamilyMembers;
  final bool isActive;
  final DateTime createdAt;

  MembershipPlan({
    required this.id,
    required this.name,
    required this.tier,
    required this.monthlyPriceCents,
    required this.yearlyPriceCents,
    required this.currency,
    required this.isFamilyPlan,
    this.maxFamilyMembers,
    required this.isActive,
    required this.createdAt,
  });

  factory MembershipPlan.fromJson(Map<String, dynamic> json) {
    return MembershipPlan(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? 'Plan Necunoscut',
      tier: json['tier'] as String? ?? 'basic',
      monthlyPriceCents: json['monthly_price_cents'] as int? ?? 0,
      yearlyPriceCents: json['yearly_price_cents'] as int? ?? 0,
      currency: json['currency'] as String? ?? 'RON',
      isFamilyPlan: json['is_family_plan'] as bool? ?? false,
      maxFamilyMembers: json['max_family_members'] as int?,
      isActive: json['is_active'] as bool? ?? true,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : DateTime.now(),
    );
  }

  double get monthlyPriceInCurrency => monthlyPriceCents / 100.0;
  double get yearlyPriceInCurrency => yearlyPriceCents / 100.0;
}

File: kratos-admin-expo/src/models/membership-plan.ts
import {
  asBoolean,
  asDate,
  flutterInteger as asInteger,
  flutterNullableInteger as asNullableInteger,
  asRecord,
  asString,
} from '@/utils/parsing';

export interface MembershipPlan {
  id: string;
  name: string;
  tier: string;
  monthlyPriceCents: number;
  yearlyPriceCents: number;
  currency: string;
  isFamilyPlan: boolean;
  maxFamilyMembers: number | null;
  isActive: boolean;
  createdAt: Date;
}

export function parseMembershipPlan(value: unknown): MembershipPlan {
  const row = asRecord(value);
  return {
    id: asString(row.id ?? row.membership_plan_id),
    name: asString(row.name ?? row.plan_name, 'Plan Necunoscut'),
    tier: asString(row.tier, 'basic'),
    monthlyPriceCents: asInteger(row.monthly_price_cents),
    yearlyPriceCents: asInteger(row.yearly_price_cents),
    currency: asString(row.currency, 'RON'),
    isFamilyPlan: asBoolean(row.is_family_plan),
    maxFamilyMembers: asNullableInteger(row.max_family_members),
    isActive: asBoolean(row.is_active, true),
    createdAt: asDate(row.created_at),
  };
}

export const monthlyPlanPrice = (plan: MembershipPlan) => plan.monthlyPriceCents / 100;
export const yearlyPlanPrice = (plan: MembershipPlan) => plan.yearlyPriceCents / 100;
